### PR TITLE
Add support for showing end location marker

### PR DESCRIPTION
**Pull request by Sija**
Code would welcome bit of DRY-ness but first I wanted to see whether this PR gain any traction.

- `DotFormatter`:

  image

- `ExplainFormatter`:

  image

### CODE REVIEW

**Review by veelenga — COMMENTED** [review]
This looks interesting. What do you think about a small adjustment to the formatting:

before:

```
a = "very long text"
     ^ ---------- ^
```

after:

```
a = "very long text"
     ^^^^^^^^^^^^^^
```

**Comment by Sija** [comment]
@veelenga I've tried but with bigger amount of occurrences it becomes an annoyance:

image

**Comment by Sija** [comment]
Preview of the current version:

image

File: src/ameba/formatter/util.cr [hunk 1]
```diff
@@ -4,42 +4,57 @@ module Ameba::Formatter
...
+        when lineno - context_lines...lineno
+          pre_context << trim(line)
+        when lineno + 1..lineno + context_lines
+          post_context << trim(line)
+        end
+      end
+
+      # remove empty lines at the beginning/end
+      pre_context.shift? unless pre_context.first?.presence
+      post_context.pop? unless post_context.last?.presence
```
Comment (veelenga) [inline src/ameba/formatter/util.cr:28]: what's the point of using `.presence` here?

File: src/ameba/reportable.cr [hunk 2]
```diff
@@ -5,35 +5,37 @@ module Ameba
     getter issues = [] of Issue
 
     # Adds a new issue to the list of issues.
-    def add_issue(rule, location : Crystal::Location?, end_location : Crystal::Location?, message, status = nil)
+    def add_issue(rule, location, end_location, message, status = nil) : Issue
```
Comment (veelenga) [inline src/ameba/reportable.cr:8]: do we rollback https://github.com/crystal-ameba/ameba/pull/184 ?

**Review by veelenga — COMMENTED** [review]
Looks great. Could you please cover this by specs, especially the new utility methods.

File: src/ameba/formatter/util.cr (hunk 1 above)
Comment (Sija) [reply src/ameba/formatter/util.cr:28]: To check for `blank?` strings.

File: src/ameba/reportable.cr (hunk 2 above)
Comment (Sija) [reply src/ameba/reportable.cr:8]: Yep, it's not needed anymoar since after this change there's no splatted arguments involved.

**Review by veelenga — APPROVED** [review]

**Pull request merged**